python/ray/autoscaler/_private/autoscaler.py

```diff
@@ -81,7 +80,8 @@ def __init__(
             update_interval_s: int = AUTOSCALER_UPDATE_INTERVAL_S,
             prefix_cluster_info: bool = False,
             event_summarizer: Optional[EventSummarizer] = None,
-            prom_metrics: Optional[AutoscalerPrometheusMetrics] = None):
+            prom_metrics: Optional[AutoscalerPrometheusMetrics] = None,
+            disable_node_updaters: bool = False):
```

Comment:
instead of a new parameter, I wonder if this can be assumed to be true if there is no SSH configs

Reply:
I think the idea would be to assume that there are no node updaters if there is no auth config. That would work and would keep monitor and autoscaler __init__ signatures cleaner.
On the other hand it could be confusing -- for example, the current logic for K8s support inserts an empty auth config to keep everything happy (and uses `kubectl exec` to execute updates.)
I think it's better for now to keep things explicit and make the init signatures uglier.

Reply:
okay, up to you. I just prefer no interface change if not necessary.

Reply:
I'll put this in the provider field of the config, since it's kind of a property of the NodeProvider. This (a) doesn't change the autoscaler/monitor Python interface (b) still makes the flag explicit.
(Open to alternatives.)